packages/core/react/src/WalletProvider.tsx

```diff
@@ -282,6 +281,13 @@ export const WalletProvider: FC<WalletProviderProps> = ({
         }
     }, [adapter, handleConnect, handleDisconnect, handleError]);
 
+    // When the adapter changes, disconnect the old one.
+    useEffect(() => {
+        return () => {
+            adapter?.disconnect();
```

Comment:
Do we need to catch this if it errors?

Reply:
I don't _think_ so, because I don't think that fatals in non-awaited promises won't bubble up through React. It will appear in the console, and in dev mode, but not in production.

https://codesandbox.io/s/musing-danny-nrpn8?file=/src/App.js